crates/sargon/src/profile/mfa/security_structures/automatic_shield_builder/automatic_shield_builder.rs

```diff
@@ -0,0 +1,916 @@
+use crate::prelude::*;
+
+use super::{
+    proto_matrix::ProtoMatrix, quantity::Quantity,
+    CallsToAssignUnsupportedFactor,
+};
+
+use FactorSourceCategory::*;
+use RoleKind::*;
+
+impl FactorSourceCategory {
+    fn is_supported(&self) -> bool {
```

Comment:
What does this mean exactly? supported as part of automatic build? so we cannot assign Custodian ever as part of automatic?